### DISCUSSION TITLE
Need information on use of EdgeX C SDK callbacks template_create_addr and template_create_resource_attr

### DISCUSSION THREAD

**Discussion by satyam1990** [edgexfoundry/community#78] [Q&A] [answered]
Hi Team,

In the Sample Application using EdgeX C SDK I am unable to figure out in which scenarios template_create_addr and template_create_resource_attr mentioned callback functions could be useful.

In the Sample application for template_create_addr function we just return the protocols as received in arguments:
```
static devsdk_address_t template_create_addr (void *impl, const devsdk_protocols *protocols, iot_data_t **exception)
{
  return (devsdk_address_t)protocols;
}
```

Similarly in callback function template_create_resource_attr we return same attribues as received as input:
```
static devsdk_resource_attr_t template_create_resource_attr (void *impl, const iot_data_t *attributes, iot_data_t **exception)
{
  return iot_data_copy (attributes);
}
```

It would be great if someone can explain with an example taking a scenario in which these callbacks will be useful. I mean what is someone supposed to do with these callbacks.

**Comment by iain-anderson** [comment, marked as answer]
Hi, there's some notes on those methods in `docs/servicewritersguide.md`. The SDK examples haven't been updated to do parsing in those callbacks, but you can see a usage example in device-bacnet-c.

**Comment by satyam1990** [comment]
Thanks @iain-anderson . There is good explanation about template_create_addr callback in the servicewritersguide.md.
And in the examples "gyro" application is making use of template_create_resource_attr callback hence that was too clear from there.

I will try out these things myself. Thanks again for your prompt help as always.
